### DISCUSSION TITLE
For guitar tabs, is PDF or epub more efficient?  Or ________?  Is supporting guitar pro files out of scope?

### DISCUSSION THREAD

**Discussion by arooni** [koreader/koreader#11331] [General]
I like to use my Kindle Scribe with KoReader at night when relaxing to see guitar tabs/music since it's so easy on the eyes.  Is one format of music easier than another?  Or is there another format that's better?

I have a whole *bunch* of guitar pro files (.gp3, .gp4, .gp5 etc).  Each is quite small individually, so they seem super efficient, and there is some open source software called TuxGuitar that could be used to read them on a regular computer, but curious if koreader could ever support this format now that the Scribe is jailbroken and such a perfect device for guitar tabs/tabulature.

**Comment by Frenzie** [comment]
I'd say that sounds a bit unlikely, but if that TuxGuitar program can run on the device you might be able to script it.

**Comment by pazos** [comment]
Pdf is way better.
Use https://gpxtopdf.com/ or similar

**Reply by Frenzie** [reply]
Huh, I wrote "script it"? To be clear I meant "script converting them to PDF on the device", heh.

**Reply by pazos** [reply]
Not sure if it's feasible to run tuxguitar on any embedded device. It requires a bunch of dependencies, most of them related to audio and midi. Also, even if possible, the print engine relies on the underlying os.

Looks much easier to just spawn a new docker stack with something like syncthing and a gpx to pdf converter that does the job and pushes the result to the consumer.

**Comment by arooni** [comment]
Why would pdf be superior to epub for this use case ?

**Comment by NiLuJe** [comment]
Because it's fixed layout? Not a musician, but that sounds sensible to me?

**Reply by pazos** [reply]
Yep, fixed layout. A tablature is a graphical representation of notes where x is time and y are strings. You want accuracy, there's nothing to reflow.

**Reply by quicklywilliam** [reply]
PDF fixes the layout at the page level, which doesn't seem like a great solution – you either need the page size to approximately match the screen size or you are stuck squinting or zooming around.

The best solution would be native support for a tab format that intelligently renders the staff to best fit the screen size. 

Perhaps second best might be a simple .tab text file (the most ubiquitous form) with some logic in reader to prevent the staff from being broken across pages.
